libyara/ahocorasick.c

```diff
@@ -152,30 +178,138 @@ static int _yr_ac_queue_is_empty(
 
 
 //
-// _yr_ac_next_state
+// _yr_ac_next_state_classes
 //
-// Given an automaton state and an input symbol, returns the new state
-// after reading the input symbol.
+// Returns true if the bitmap1 and bitmap2 are the same.
+//
+
+static bool _yr_ac_compare_classes(
+    YR_BITMASK* bitmap1,
+    YR_BITMASK* bitmap2)
+{
+  for (int i = 0; i < YR_BITMAP_SIZE; i++)
```

Comment:
What about using memcmp? memcmp is usually highly optimized and can leverage SSE instructions.